src/sc/sidechain.cpp

```diff
@@ -713,17 +713,30 @@ int CSidechain::getNumBlocksForScFeeCheck()
 
 int CSidechain::getMaxSizeOfScFeesContainers()
 {
-    int epochLength = fixedParams.withdrawalEpochLength;
-    int numBlocks = getNumBlocksForScFeeCheck();
-
-    //assert(epochLength > 0);
-    if (epochLength == 0) {
-        maxSizeOfScFeesContainers = 1;
-    } else {
-        maxSizeOfScFeesContainers = numBlocks / epochLength;
-        if (maxSizeOfScFeesContainers == 0 || (numBlocks % epochLength != 0) )
-        {
-            maxSizeOfScFeesContainers++;
+    if (maxSizeOfScFeesContainers == -1) {
+        maxSizeOfScFeesContainers = getNumBlocksForScFeeCheck();
+
+        if (!isNonCeasing()) {
+            int epochLength = fixedParams.withdrawalEpochLength;
+            int numBlocks = getNumBlocksForScFeeCheck();
+            if (epochLength == 0) {
+                maxSizeOfScFeesContainers = 1;
+            } else {
+                maxSizeOfScFeesContainers = numBlocks / epochLength;
+                if (maxSizeOfScFeesContainers == 0 || (numBlocks % epochLength != 0) )
+                {
+                    maxSizeOfScFeesContainers++;
+                }
+            }
+        }
+        else {
+            // As defined in chainparams.cpp, maxSizeOfScFeesContainers is 200 for mainnet and testnet,
+            // and 10 for regtest. For regtest, this value can be overridden with the command line option
+            // -blocksforscfeecheck=<n>. In this case, we provide a safe guard of five slots for the scFees
+            // size. This value has been chosen as it was the max possible size obtainable by the previous
+            // implementation.
+            if (maxSizeOfScFeesContainers < 5)
```

Comment:
I find semantically better 
```
maxSizeOfScFeesContainers = std::max(maxSizeOfScFeesContainers, 5)
```

Reply:
Also ... what is that magic constant 5 ? Can we give a meaningful value name using `static constexpr` somewhere ?

Reply:
The value 5 is derived from the default values set in `chainparams.cpp`. Specifically, for regtest `nScNumBlocksForScFeeCheck` (the values returned by `CSidechain::getNumBlocksForScFeeCheck()`) is set to 10, and the withdrawal epoch length can range from 2 to 4032 (`nScMinWithdrawalEpochLength` and `nScMaxWithdrawalEpochLength` also in `chainparams.cpp`).

These values allow the `scFees` list size to range from 1 to 5, as evaluated by `numBlocks / epochLength` (and the subsequent `if` block).

I was not sure whether all these details were meaningful, so I skipped them in the comment altogether, but I realize that by looking at the value it is not clear where it comes from. I doubt that a meaningful name could help here, so instead I suggest I could update the comment, so that its meaning should not be lost.

Reply:
Great insight .... thank you. 
So let me ask ... can't we have a function deriving the max value instead of setting as a constant ?

Reply:
`if..then` replaced by `std::max`, and comment improved in f0bab99cd7559a9bab4c8dc65103df6eea2b4b76